### DISCUSSION TITLE
`FromRequest` is not implemented for `BearerAuth`

### DISCUSSION THREAD

**Discussion by HerrMuellerluedenscheid** [actix/actix-extras#310] [Q&A] [answered]
Hi,

I tried the provided [example](https://docs.rs/actix-web-httpauth/latest/actix_web_httpauth/extractors/bearer/struct.BearerAuth.html#examples-1) and got the following error which is strange because I can see that `FromRequest` is implemented. Do you have a hint for me what is wrong?
 
```
error[E0277]: the trait bound `BearerAuth: FromRequest` is not satisfied
   --> src/main.rs:15:64
    |
15  |         .service(web::resource("/index.html").route(web::get().to(index)));
    |                                                                ^^ the trait `FromRequest` is not implemented for `BearerAuth`
```

BTW: I tried to access the gitter chat but that gave me a 404 :/

**Comment by robjtede** [comment, marked as answer]
My guess is that you're using an incompatible version of Actix Web, maybe v3. All latest crates use v4.

**Reply by HerrMuellerluedenscheid** [reply]
Yes, that was the problem! Thanks for reaching out so incredibly quickly!
